api/src/clojure/org/akvo/flow_api/unilog/unilog.clj

```diff
@@ -104,32 +105,32 @@
       []
       reducible)))
 
-(defn after-forms-loaded [{::keys [forms-instances-grouped-by-form
-                                   unilog-id
-                                   form-updated
-                                   form-deleted
-                                   form-instance-deleted
-                                   data-point-changed
-                                   data-point-deleted
-                                   survey-changed
-                                   survey-deleted]}
-                          form-id->form]
+(defn filter-events-by-authorization [{::keys [forms-instances-grouped-by-form
+                                               unilog-id
+                                               form-updated
+                                               form-deleted
+                                               form-instance-deleted
+                                               data-point-changed
+                                               data-point-deleted
+                                               survey-changed
+                                               survey-deleted]}
+                                      form-id->form
+                                      authorized-forms
+                                      authorized-surveys]
   {:unilog-id unilog-id
-   :form-changed (->> form-updated
-                   (keep form-id->form)
-                   set)
-   :form-deleted form-deleted
+   :form-changed (->> form-updated (keep form-id->form) set)
+   :form-deleted (set/intersection authorized-forms form-deleted)
```

Comment:
A deleted form will never ever be in the authorised-forms, so the :form-deleted will always always be the empty set.

Deleted forms cannot be authorised. Should be made clear in the API doc that any delete may contain objects that the client does not know about.

Note that the end to end tests are lying here, as they are not properly simulating the whole flow, as they do not delete the objects in the datastore even if they put a "delete" message in the "fake" unilog DB.

Reply:
> A deleted form will never ever be in the authorised-forms, so the :form-deleted will always always be the empty set.

You're right and I thought about it. When we find a `form-deleted` the form is already gone, so looking for authorized forms will not give us that id. :thinking:

Reply:
> Note that the end to end tests are lying here, as they are not properly simulating the whole flow, as they do not delete the objects in the datastore even if they put a "delete" message in the "fake" unilog DB.

Correct. You're right.